library/std/src/io/mod.rs

```diff
@@ -917,6 +917,17 @@ pub trait Read {
     /// # }
     /// ```
     ///
+    /// # Usage Notes
+    ///
+    /// `read_to_end` attempts to read a source until EOF, but many sources are continuous streams
+    /// that do not send EOF. In these cases, `read_to_end` will block indefinitely. Standard input
+    /// is one such stream which may be finite if piped, but is typically continuous. For example,
+    /// `cat <file> | <my_rust_program>` will correctly terminate with an `EOF` upon closure of cat.
+    /// `yes "Data" | pv | <my_rust_program` or `tail -f <file> | <my_rust_program>` may not close
+    /// the stream or insert an `EOF` termination character.
```

Comment:
This looks much better to me. I think the `cat` example would be good to keep, but maybe translate the other example into words in case someone doesn't know what the `-f` flag does

```rust
    /// `read_to_end` attempts to read a source until EOF, but many sources are continuous streams
    /// that do not send EOF. In these cases, `read_to_end` will block indefinitely. Standard input
    /// is one such stream which may be finite if piped from a finite source (e.g. `cat foo.txt |
    /// my-program`), but is typically continuous (reading user input or sources that stay open).
```